tests/deploy/test_nemo_deploy.py

```diff
@@ -0,0 +1,302 @@
+# Copyright (c) 2024, NVIDIA CORPORATION.  All rights reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import json
+import os
+import shutil
+import time
+import urllib.request as req
+from pathlib import Path
+
+import numpy as np
+import pytest
+import torch
+from tqdm import tqdm
+
+from nemo.deploy import DeployPyTriton, DeployTensorRTLLM, NemoQuery, NemoQueryTensorRTLLM
+from nemo.export import TensorRTLLM
+from tests.infer_data_path import download_nemo_checkpoint, get_infer_test_data
+
```

Comment:
Thank you for adding test, however unit tests are meant to be run on small models and take less than 10-20 seconds each - and export of such large models is not in the scope of unit tests but rather Jenkins tests.

Furthermore, it requires many difficult libraries to be pre-installed - such as trtllm and pytriton. These libraries are not part of the requirements by default and will fail at import level (due to above import time raising of error).

@ericharper what do you think of moving this to jenkins / actions?

Reply:
Well, there is unfortunately no way to run any export to trt llm test within 10-20 seconds. The fastest one with a very small LLM model will take at least 2-3 minutes. So, if this is a hard requirement, then we don't have to include it in the NeMo github CI. But we should keep this file since we use it for the nemo fw container CI. 

And this file won't be run with the pytest command since it's a script.

@ericharper @titu1994 Please let me know if you guys don't want this test in github CI. I added it in the jenkinsfile and I can remove it.

Reply:
@ericharper we have nightly tests now I think, best to keep such long running tests there ?

Reply:
Removed the test from Jenkinsfile. We'll use nemo fw container CI.